frontend/public/components/build-logs.jsx

```diff
@@ -27,25 +35,32 @@ export class BuildLogs extends React.Component {
     };
   }
 
-  static getDerivedStateFromProps(nextProps, prevState) {
-    const phase = _.get(nextProps.obj, 'status.phase');
+  static getDerivedStateFromProps({ obj: build }, { status: prevStatus }) {
+    const phase = _.get(build, 'status.phase');
     const status = buildToLogSourceStatus(phase);
-    if (prevState.status !== status){
+    if (prevStatus !== status){
       return {status};
     }
     return null;
   }
 
   render() {
-    const namespace = _.get(this.props.obj, 'metadata.namespace');
-    const buildName = _.get(this.props.obj, 'metadata.name');
+    const { obj: build } = this.props;
+    const namespace = _.get(build, 'metadata.namespace');
+    const buildName = _.get(build, 'metadata.name');
+    const isPipeline = _.get(build, 'spec.strategy.type') === 'JenkinsPipeline';
+    const jenkinsLink = _.get(build, ['metadata', 'annotations', 'openshift.io/jenkins-log-url']);
```

Comment:
You might export the helpers from build-pipeline.tsx so we don't have this in two places